com.playeveryware.eos/Runtime/Core/EOSManager.cs

```diff
@@ -138,6 +135,40 @@ public delegate void OnAuthLinkExternalAccountCallback(
         //cached log levels for retrieving later
         private static Dictionary<LogCategory, LogLevel> logLevels;
 
+        /// <summary>
+        /// Lazy-loaded singleton reference for the EOSManager.
+        /// If it doesn't exist when queried, it will create a GameObject,
+        /// and add an EOSManager component to it.
+        /// This can serve as an alternate way of setting up EOSManager, rather
+        /// than using a prefab with the script on it.
+        /// </summary>
+        private static Lazy<EOSManager> s_lazyManagerInstance = new Lazy<EOSManager>(() =>
+        {
+            EOSSingleton.print($"{nameof(EOSManager)} ({nameof(s_lazyManagerInstance)}): Lazy loaded singleton instance is creating a GameObject with an EOSManager attached.");
+
+            // EOSManager is a MonoBehaviour, so create a GameObject to hold it
+            GameObject instanceGameObjectHolder = new GameObject();
+
+            // Add the component
+            EOSManager addedManager = instanceGameObjectHolder.AddComponent<EOSManager>();
+
+            return addedManager;
+        });
+
+        public static EOSManager ManagerInstance
```

Comment:
I understand why you chose to name this `ManagerInstance` instead of `Instance`, but I think doing so reinforces a bad pattern that we are currently using. 

`EOSSingleton` is a bad name for a variety of reasons:
* Singletons should not contain the word `Singleton`
* Most of the functionality within `EOSManager` exists inside `EOSSingleton`
* There can be multiple instances of `EOSSingleton` (that is, there is nothing preventing anyone from creating more instances of it)

Ostensibly - despite utilizing vocabulary often associated with the singleton pattern - _neither_ `EOSManager` nor `EOSSingleton` _are_ following the singleton pattern.

The Correct™ way for this to be implemented is to move all the functions within `EOSSingleton` into `EOSManager`, and simply have `EOSManager` use the singleton pattern. 

Now, that is clearly outside of the scope of this PR but beginning to implement `ManagerInstance` _reinforces_ that poor implementation and will make it difficult in the future to correct properly.

Reply:
We talked in Slack about this; Ultimately this is right, we don't want to entrench ourselves further in to bad patterns.